storage/cache/flag.go

```diff
@@ -7,11 +7,26 @@ import (
 	flipt "github.com/markphelps/flipt/rpc"
 	"github.com/markphelps/flipt/storage"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/sirupsen/logrus"
 )
 
 const flagCachePrefix = "flag:"
 
+var (
+	cacheHitTotal = promauto.NewCounter(prometheus.CounterOpts{
+		Name:        "cacheHitTotal",
+		ConstLabels: prometheus.Labels{"type": "flag", "cache": "memory"},
```

Comment:
I dont think we want to use `ConstLabels` here, as that means those labels will always be applied and wont allow us to override them, for example for other `type:` labels, (like `evaluation`).

From the prom [go docs](https://godoc.org/github.com/prometheus/client_golang/prometheus#Opts):

>  // ConstLabels are used to attach fixed labels to this metric. Metrics
    // with the same fully-qualified name must have the same label names in
    // their ConstLabels.
    //
    // ConstLabels are only used rarely.

Instead, I think we want to use the labels at time of use, so that `cacheHitTotal` and `cacheMissTotal` will be available package wide, but when you use them you'll add the labels, ex:

```go
cacheHitTotal.WithLabelValues("type", "flag", "cache", "memory").Inc()
```

Then for when we add evaluation cache, it will be like:

```go
cacheHitTotal.WithLabelValues("type", "evaluation", "cache", "memory").Inc()
```

See: https://godoc.org/github.com/prometheus/client_golang/prometheus#CounterVec.WithLabelValues

Reply:
That sounds great to me.  Any thoughts on using [with](https://github.com/prometheus/client_golang/blob/master/prometheus/counter.go#L214) instead? With would allow us to set the label name and value rather than just setting label values.

Reply:
Please see the updates @markphelps!

Also I'm still looking into the best way to handle the histograms. Was leaning towards an approach similar to one prometheus takes [here](https://github.com/prometheus/prometheus/blob/851131b0740be7291b98f295567a97f32fffc655/web/web.go#L98).